src/orchestrator-kubernetes/src/lib.rs

```diff
@@ -223,6 +224,8 @@ impl NamespacedOrchestrator for NamespacedKubernetesOrchestrator {
             .iter()
             .map(|p| (p.name.clone(), p.port_hint))
             .collect();
+        let node_selector = availability_zone
+            .map(|az| BTreeMap::from([("topology.kubernetes.io/zone".to_string(), az)]));
```

Comment:
I am pretty sure that this is not correct. [As the design doc notes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/20220413_cluster_replica.md#orchestrator-changes), availability zones must be specified by their IDs, not their account-specific names. According to https://github.com/kubernetes/cloud-provider-aws/issues/300, AWS sets `topology.kubernetes.io/zone` to the AZ name and there is no alternative label for the AZ ID.

Would it be possible to introduce a Materialize-specific label with the AZ ID instead?

Reply:
Relevant Slack discussion: https://materializeinc.slack.com/archives/CL68GT3AT/p1650458520034059

Reply:
> Would it be possible to introduce a Materialize-specific label with the AZ ID instead?

Ah, so this is exactly what I was thinking! I think we should introduce our _own_ label here, regardless of what approach to IDs/names we take.

I recommend using the name `materialize.cloud/availability-zone`. Then MaterializeInc/cloud is wholly responsible for attaching that label to nodes however it likes.